src/runtime/slot-polyfill-utils.ts

```diff
@@ -190,4 +193,89 @@ export const addSlotRelocateNode = (
 };
 
 export const getSlotName = (node: d.PatchedSlotNode) =>
-  node['s-sn'] || (node.nodeType === 1 && (node as Element).getAttribute('slot')) || '';
+  typeof node['s-sn'] === 'string'
+    ? node['s-sn']
+    : (node.nodeType === 1 && (node as Element).getAttribute('slot')) || undefined;
+
+/**
+ * Add `assignedElements` and `assignedNodes` methods on a fake slot node
+ *
+ * @param node - slot node to patch
+ */
+export function patchSlotNode(node: d.RenderNode) {
+  if ((node as any).assignedElements || (node as any).assignedNodes || !node['s-sr']) return;
+
+  const assignedFactory = (elementsOnly: boolean) =>
+    function (opts?: { flatten: boolean }) {
+      const toReturn: d.RenderNode[] = [];
+      const slotNamesToFind = [this['s-sn']];
+
+      if ((this['s-sn'], opts?.flatten && this.nodeType === NODE_TYPE.ElementNode && this.childNodes.length)) {
+        // if we're flattening, we need to find all nested <slot /> nodes
+        const getNestedSlotNames = (slot: d.RenderNode) => {
+          (slot.childNodes as NodeListOf<d.RenderNode>).forEach((nestedNode) => {
+            if (nestedNode['s-sr']) {
+              // found a slot node. Let's add it to the list of slot names to find
+              slotNamesToFind.push(getSlotName(nestedNode));
+              if (nestedNode.nodeType === NODE_TYPE.ElementNode && !nestedNode.hidden) {
+                // this 'slot' is also a `<slot-fb />` so we need to drill down recursively
+                getNestedSlotNames(nestedNode);
+              }
+            } else {
+              toReturn.push(nestedNode as d.RenderNode);
+            }
+          });
+        };
+        getNestedSlotNames(this);
+      }
```

Comment:
not sure if we need to support `{deep: true}`? 
Quite a niche / esoteric thing perhaps and not worth the LOC... WDYT?

Reply:
yep - for sure